converters/toZigbee.js

```diff
@@ -2387,6 +2387,28 @@ const converters = {
             await tuya.sendDataPointValue(entity, dp, newValue, 'setData', 1);
         },
     },
+    tuya_switch_state_multi: {
```

Comment:
I don't expect this converter to be needed, can you provide the debug log when using the `tuya_switch_state` converter?

See https://www.zigbee2mqtt.io/information/debug.html on how to enable debug logging.

Reply:
even better... I even added more console log on the `tuya_switch_state` and printed all the arguments received:
```
May 07 10:08:22 raspberrypi npm[3529]: Zigbee2MQTT:debug 2021-05-07 10:08:22: Received MQTT message on 'zigbee2mqtt/0x60a423fffeb29907/set' with data '{"state_l2":"ON","state_l1":"ON"}'
May 07 10:08:22 raspberrypi npm[3529]: Zigbee2MQTT:debug 2021-05-07 10:08:22: Publishing 'set' 'state' to 'Kitchen Lights'
May 07 10:08:22 raspberrypi npm[3529]: {
May 07 10:08:22 raspberrypi npm[3529]:   entity: Endpoint {
May 07 10:08:22 raspberrypi npm[3529]:     ID: 1,
May 07 10:08:22 raspberrypi npm[3529]:     profileID: 260,
May 07 10:08:22 raspberrypi npm[3529]:     deviceID: 81,
May 07 10:08:22 raspberrypi npm[3529]:     inputClusters: [ 0, 4, 5, 61184 ],
May 07 10:08:22 raspberrypi npm[3529]:     outputClusters: [ 25, 10 ],
May 07 10:08:22 raspberrypi npm[3529]:     deviceNetworkAddress: 54347,
May 07 10:08:22 raspberrypi npm[3529]:     deviceIeeeAddress: '0x60a423fffeb29907',
May 07 10:08:22 raspberrypi npm[3529]:     clusters: { genBasic: [Object] },
May 07 10:08:22 raspberrypi npm[3529]:     _binds: [ [Object] ],
May 07 10:08:22 raspberrypi npm[3529]:     _configuredReportings: [],
May 07 10:08:22 raspberrypi npm[3529]:     meta: {}
May 07 10:08:22 raspberrypi npm[3529]:   },
May 07 10:08:22 raspberrypi npm[3529]:   key: 'state',
May 07 10:08:22 raspberrypi npm[3529]:   value: 'ON',
May 07 10:08:22 raspberrypi npm[3529]:   meta: {
May 07 10:08:22 raspberrypi npm[3529]:     endpoint_name: 'l2',
May 07 10:08:22 raspberrypi npm[3529]:     options: {
May 07 10:08:22 raspberrypi npm[3529]:       friendly_name: 'Kitchen Lights',
May 07 10:08:22 raspberrypi npm[3529]:       ID: '0x60a423fffeb29907',
May 07 10:08:22 raspberrypi npm[3529]:       friendlyName: 'Kitchen Lights',
May 07 10:08:22 raspberrypi npm[3529]:       type: 'device'
May 07 10:08:22 raspberrypi npm[3529]:     },
May 07 10:08:22 raspberrypi npm[3529]:     message: { state_l1: 'ON', state: 'ON' },
May 07 10:08:22 raspberrypi npm[3529]:     logger: DerivedLogger {
May 07 10:08:22 raspberrypi npm[3529]:       _readableState: [ReadableState],
May 07 10:08:22 raspberrypi npm[3529]:       readable: true,
May 07 10:08:22 raspberrypi npm[3529]:       _events: [Object: null prototype],
May 07 10:08:22 raspberrypi npm[3529]:       _eventsCount: 3,
May 07 10:08:22 raspberrypi npm[3529]:       _maxListeners: undefined,
May 07 10:08:22 raspberrypi npm[3529]:       _writableState: [WritableState],
May 07 10:08:22 raspberrypi npm[3529]:       writable: true,
May 07 10:08:22 raspberrypi npm[3529]:       allowHalfOpen: true,
May 07 10:08:22 raspberrypi npm[3529]:       _transformState: [Object],
May 07 10:08:22 raspberrypi npm[3529]:       silent: undefined,
May 07 10:08:22 raspberrypi npm[3529]:       format: [Format],
May 07 10:08:22 raspberrypi npm[3529]:       defaultMeta: null,
May 07 10:08:22 raspberrypi npm[3529]:       levels: [Object],
May 07 10:08:22 raspberrypi npm[3529]:       level: 'info',
May 07 10:08:22 raspberrypi npm[3529]:       exceptions: [ExceptionHandler],
May 07 10:08:22 raspberrypi npm[3529]:       rejections: [RejectionHandler],
May 07 10:08:22 raspberrypi npm[3529]:       profilers: {},
May 07 10:08:22 raspberrypi npm[3529]:       exitOnError: true,
May 07 10:08:22 raspberrypi npm[3529]:       addTransport: [Function (anonymous)],
May 07 10:08:22 raspberrypi npm[3529]:       cleanup: [Function: cleanup],
May 07 10:08:22 raspberrypi npm[3529]:       logOutput: [Function: logOutput],
May 07 10:08:22 raspberrypi npm[3529]:       getLevel: [Function (anonymous)],
May 07 10:08:22 raspberrypi npm[3529]:       setLevel: [Function (anonymous)],
May 07 10:08:22 raspberrypi npm[3529]:       warn: [Function (anonymous)],
May 07 10:08:22 raspberrypi npm[3529]:       [Symbol(kCapture)]: false
May 07 10:08:22 raspberrypi npm[3529]:     },
May 07 10:08:22 raspberrypi npm[3529]:     device: Device {
May 07 10:08:22 raspberrypi npm[3529]:       ID: 7,
May 07 10:08:22 raspberrypi npm[3529]:       _type: 'EndDevice',
May 07 10:08:22 raspberrypi npm[3529]:       _ieeeAddr: '0x60a423fffeb29907',
May 07 10:08:22 raspberrypi npm[3529]:       _networkAddress: 54347,
May 07 10:08:22 raspberrypi npm[3529]:       _manufacturerID: 4098,
May 07 10:08:22 raspberrypi npm[3529]:       _endpoints: [Array],
May 07 10:08:22 raspberrypi npm[3529]:       _manufacturerName: '_TZE200_g1ib5ldv',
May 07 10:08:22 raspberrypi npm[3529]:       _powerSource: 'Mains (single phase)',
May 07 10:08:22 raspberrypi npm[3529]:       _modelID: 'TS0601',
May 07 10:08:22 raspberrypi npm[3529]:       _applicationVersion: 66,
May 07 10:08:22 raspberrypi npm[3529]:       _stackVersion: 0,
May 07 10:08:22 raspberrypi npm[3529]:       _zclVersion: 3,
May 07 10:08:22 raspberrypi npm[3529]:       _hardwareVersion: 1,
May 07 10:08:22 raspberrypi npm[3529]:       _dateCode: '',
May 07 10:08:22 raspberrypi npm[3529]:       _softwareBuildID: undefined,
May 07 10:08:22 raspberrypi npm[3529]:       _interviewCompleted: true,
May 07 10:08:22 raspberrypi npm[3529]:       _interviewing: false,
May 07 10:08:22 raspberrypi npm[3529]:       _skipDefaultResponse: false,
May 07 10:08:22 raspberrypi npm[3529]:       meta: [Object],
May 07 10:08:22 raspberrypi npm[3529]:       _lastSeen: 1620371299020,
May 07 10:08:22 raspberrypi npm[3529]:       _linkquality: 70
May 07 10:08:22 raspberrypi npm[3529]:     },
May 07 10:08:22 raspberrypi npm[3529]:     state: { state_l1: 'OFF', state_l2: 'OFF' },
May 07 10:08:22 raspberrypi npm[3529]:     membersState: null,
May 07 10:08:22 raspberrypi npm[3529]:     mapped: {
May 07 10:08:22 raspberrypi npm[3529]:       fingerprint: [Array],
May 07 10:08:22 raspberrypi npm[3529]:       model: 'ZTS-EU_2gang',
May 07 10:08:22 raspberrypi npm[3529]:       vendor: 'Moes',
May 07 10:08:22 raspberrypi npm[3529]:       description: 'Wall touch light switch (2 gang)',
May 07 10:08:22 raspberrypi npm[3529]:       exposes: [Array],
May 07 10:08:22 raspberrypi npm[3529]:       fromZigbee: [Array],
May 07 10:08:22 raspberrypi npm[3529]:       toZigbee: [Array],
May 07 10:08:22 raspberrypi npm[3529]:       meta: [Object],
May 07 10:08:22 raspberrypi npm[3529]:       endpoint: [Function: endpoint],
May 07 10:08:22 raspberrypi npm[3529]:       configure: [AsyncFunction: configure]
May 07 10:08:22 raspberrypi npm[3529]:     }
May 07 10:08:22 raspberrypi npm[3529]:   }
May 07 10:08:22 raspberrypi npm[3529]: }
May 07 10:08:22 raspberrypi npm[3529]: Zigbee2MQTT:info  2021-05-07 10:08:22: MQTT publish: topic 'zigbee2mqtt/Kitchen Lights', payload '{"linkquality":70,"state_l1":"OFF","state_l2":"ON"}'
May 07 10:08:22 raspberrypi npm[3529]: Zigbee2MQTT:debug 2021-05-07 10:08:22: Received Zigbee message from 'Kitchen Lights', type 'commandGetData', cluster 'manuSpecificTuya', data '{"data":{"data":[1],"type":"Buffer"},"datatype":1,"dp":2,"fn":0,"status":0,"transid":2}' from endpoint 1 with groupID 0
May 07 10:08:22 raspberrypi npm[3529]: Zigbee2MQTT:info  2021-05-07 10:08:22: MQTT publish: topic 'zigbee2mqtt/Kitchen Lights', payload '{"linkquality":68,"state_l1":"OFF","state_l2":"ON"}'
```

as you can see, the message from homebridge-z2m is sending ON command for both (in a scene) l1 and l2: `{"state_l2":"ON","state_l1":"ON"}`
the meta in the converter shows only one endpoint is being handled (l2), state_l2 is being replaced in the message with "state", and the converter only handles 1 of the states in the message.

if you have any suggestion I'll be happy to hear about it and modify or remove the new converter.

Reply:
Found the issue, z2m did not do multiple publishes when the endpoint IDs are the same (since normally l1 and l2 would be a different endpoint number). Fixed this now in the dev branch.(https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

Reply:
I have removed the new converter since it's not longer needed thanks to your fix